### ISSUE TITLE
Add support for hardware performance counters

### ISSUE THREAD

**Issue by wahez** [google/benchmark#869]
A benchmark can show you if some code is fast or slow. Having performance counters running automatically during a benchmark can also tell you why it is slow.

**Comment by LebedevRI** [comment]
> Having performance counters running automatically during a benchmark can also tell you why it is slow.

Can you expand that thought please?
Right away i can't recall which of those counters would always tell *why* it is slow.

**Comment by wahez** [comment]
Good point. Let me rephrase that: It will help you find out why it is slow (by allowing you to measure cache misses, instruction count, etc).

**Comment by dominichamon** [comment]
is this something you could do with custom user counters? it might be worth a prototype that does it in "user space" (ie, in the benchmark) to show the value and then we could promote it into the library.

**Comment by dominichamon** [comment]
now available thanks to #1114
